tasks/libs/common/omnibus.py

```diff
@@ -202,22 +202,27 @@ def env_filter(item):
     return dict(filter(env_filter, sorted(os.environ.items())))
 
 
-def _last_omnibus_changes(ctx):
-    omnibus_invalidating_files = ['omnibus/config/', 'omnibus/lib/', 'omnibus/omnibus.rb']
-    omnibus_last_commit = ctx.run(
-        f'git log -n 1 --pretty=format:%H {" ".join(omnibus_invalidating_files)}', hide='stdout'
-    ).stdout
-    # The commit sha1 is likely to change between a PR and its merge to main
-    # In order to work around this, we hash the commit diff so that the result
-    # can be reproduced on different branches with different sha1
-    omnibus_last_changes = ctx.run(
-        f'git diff {omnibus_last_commit}~ {omnibus_last_commit} {" ".join(omnibus_invalidating_files)}', hide='stdout'
-    ).stdout
-    hash = hashlib.sha1()
-    hash.update(str.encode(omnibus_last_changes))
-    result = hash.hexdigest()
-    print(f'Hash for last omnibus changes is {result}')
-    return result
+def _hash_paths(hasher, paths: list[str]):
+    """Update hashlib.hash object `hasher` by recursive hashing of the contents provided in `paths`."""
+
+    def hash_file(filepath):
+        with open(filepath, 'rb') as f:
```

Comment:
shouldn't we add the (relative) path itself? (to track file moves)

Reply:
The idea being that a file being renamed without changed contents (and still being in the same order after sorting) would give us the same hash? Sounds unlikely but indeed it doesn't hurt to account for it.

Reply:
Added this in [3b0aeae](https://github.com/DataDog/datadog-agent/pull/39497/commits/3b0aeaeef0ee876c7b6615548f1c3ab310ce2c5b).